packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -553,8 +558,26 @@ export function getValidatorApi({
       const blockValueEngine = enginePayloadValue + gweiToWei(consensusBlockValueEngine); // Total block value is in wei
 
       let executionPayloadSource: ProducedBlockSource | null = null;
+      const shouldOverrideBuilder = fullBlock?.shouldOverrideBuilder ?? false;
+
+      // handle the builder override case separately
+      if (shouldOverrideBuilder === true) {
+        // this is just to make typescript happy as shouldOverrideBuilder can be true only will valid
+        // full block response
+        if (fullBlock === null) {
+          throw Error("Invalid null fullBlock with builder override");
+        }
 
-      if (fullBlock && blindedBlock) {
+        executionPayloadSource = ProducedBlockSource.engine;
+        logger.verbose("Selected engine block as censorship suspected in builder blocks", {
+          // winston logger doesn't like bigint
+          enginePayloadValue: `${enginePayloadValue}`,
+          consensusBlockValueEngine: `${consensusBlockValueEngine}`,
+          blockValueEngine: `${blockValueEngine}`,
+          slot,
+          shouldOverrideBuilder,
```

Comment:
Logging the slot in the middle of the block building values looks wrong. Might be better to have it at the beginning or end. I think we have the slot as the first part of most logs